source/concurrent/Promise.ooc

```diff
@@ -71,18 +71,19 @@ ConditionPromise: class extends Promise {
 		super()
 	}
 	signal: func {
-		this _mutex with(||
-			this _completed = true
-			this _condition broadcast()
-		)
+		this _mutex lock()
+		this _completed = true
+		this _condition broadcast()
+		this _mutex unlock()
```

Comment:
This one puzzles me. When using `with()`, one of the tests in other repo reports more race conditions.
But the main reason I changed - I think it's not worth creating / deleting a temporary closure object with `calloc / free` just to save few characters (not even a line).